app/models/factory.rb

```diff
@@ -80,4 +80,31 @@ def depth
       asso.factory.depth
     end.max + 1
   end
+
+  # Return factory's familty array. index equal to depth.
+  def family
+    queue = []
+    queue.unshift(self)
+    results = []
+    generation = 0
+    while (!queue.empty?)
+      count = queue.inject(0) { |sum, factory| sum + factory.assos.size }
+      break if count == 0
+      results[generation] = []
+      while (count > 0)
```

Comment:
Style/ParenthesesAroundCondition: Don't use parentheses around the condition of a `while`.